gensim/models/tfidfmodel.py

```diff
@@ -144,17 +262,27 @@ def __getitem__(self, bow, eps=1e-12):
 
         # unknown (new) terms will be given zero weight (NOT infinity/huge weight,
         # as strict application of the IDF formula would dictate)
+
+        termid_array, tf_array = [], []
+        for termid, tf in bow:
+            termid_array.append(termid)
+            tf_array.append(tf)
+
+        tf_array = self.wlocal(np.array(tf_array))
+
         vector = [
-            (termid, self.wlocal(tf) * self.idfs.get(termid))
-            for termid, tf in bow if self.idfs.get(termid, 0.0) != 0.0
+            (termid, tf * self.idfs.get(termid))
```

Comment:
@menshikh-iv can we have something like `(termid, self.normalize(tf * self.idfs.get(termid)))` instead? So that the user can initialize with something like `normalize=np.log` ? Because right now the self.normalize function should be like `lambda tf, wt : tf, np.log(wt)` .